pkg/test/istioio/framework.go

```diff
@@ -162,19 +155,48 @@ func checkFile(path string) (*TestCase, error) {
 	config := setups[0][1]
 
 	// Check for proper test cleanup
-	testScript = createClusterSnapshots + testScript
-	cleanupScript += clusterCleanupCheck
+	scriptPrefix := getTemplateScript(scriptPrefixTemplate, shortPath)
+	testScript = addScriptPrefix(scriptPrefix, testScript)
+	cleanupScript = addScriptPrefix(scriptPrefix, cleanupScript)
 
 	testCase = &TestCase{
 		valid:         true,
 		path:          shortPath,
 		config:        config,
-		testScript:    helperScript + testScript,
-		cleanupScript: helperScript + cleanupScript,
+		testScript:    testScript,
+		cleanupScript: cleanupScript,
 	}
 	return testCase, nil
 }
 
+func addScriptPrefix(prefix, script string) string {
+	out := ""
+	needToAdd := true
+
+	// Add the prefix before the first uncommented line.
+	for i, line := range strings.Split(script, "\n") {
+		// First check if the script doesn't start with the shell line.
+		// In that case, just start the output with a default shell line and
+		// the prefix.
+		if i == 0 && !strings.HasPrefix(line, "#!") {
+			//
+			out = fmt.Sprintf("#!/usr/bin/env bash\n%s\n%s", prefix, out)
```

Comment:
Why do we want to hide this special shebang injection here? Shouldn't it be required on line 1 of any script we call this function with, or otherwise not require it so we can ignore the issue?

Reply:
done.